api/pom.xml

```diff
@@ -19,7 +19,7 @@
     <parent>
         <groupId>org.eclipse.microprofile.metrics</groupId>
         <artifactId>microprofile-metrics-parent</artifactId>
-        <version>5.1.2-SNAPSHOT</version>
+        <version>5.2.0-SNAPSHOT</version>
```

Comment:
If the PR is changing the version to 5.2.0 to avoid the `bnd baseline` failures I was seeing in my local change (I was trying to use `5.1.2-SNAPSHOT`) I understand these changes, but the change to the `Snapshot` abstract class is not a functional addition over 5.1.1 as true semantic version would imply. 

Releasing the change to `Snapshot` as 5.1.2 would be more intuitive to our consumers, but it seems the PR would have to add a `bnd baseline` exclusion for the updated method. I was not able to look into that. Perhaps @Channyboy can help here?